### ISSUE TITLE
Fresco do not display after grant permission

### ISSUE THREAD

**Issue by vinhnguyenitvn** [facebook/fresco#2159]
I have a problem when using Fresco to show an external uri

Open app and enter an activity without READ_STORAGE_PERMISSION
Ask for that permission with ActivityCompat

After get the result (the permission granted), I collect all images from external content uri and get the uris like this: content://media/external/images/media/125 
but fresco give error:
AbstractDraweeController: controller e65ea68 1: final_failed @ onFailure: failure: java.io.FileNotFoundException: /storage/emulated/0/DCIM/Camera/IMG_20180717_182941.jpg (Permission denied)

Force close app and open again (now app has permission), the image I collected display normally

* Fresco version: 1.9.0 on gradle
* Platform version: Android target 28, emulator

**Comment by zmroczek** [comment]
Hi @vinhnguyenitvn

Thanks for reporting the issue. Could you please provide a sample code that will let us reproduce this issue?

**Comment by vinhnguyenitvn** [comment]
Hi @zmroczek,
I added the code in the post, could you check?

**Comment by vinhnguyenitvn** [comment]
I tested again and it happens in just Android emulator

**Comment by lambdapioneer** [comment]
So, if I understand correctly, this is what you would expect here:

 - App starts without permission (and thus we couldn't display any image)
 - Then you ask for permission and if granted you issue an image request
 - Then it seems that the permission didn't properly propagate and the subsequent image request fails.

I'll mark this as a bug as I'd also expect that image requests would work then. Adding @oprisnik for how his view on expected behaviour.

@vinhnguyenitvn can you check what happens if all Fresco initialization code also goes after the permission grant (i.e. in the success branch of `onRequestPermissionsResult`)?

**Comment by oprisnik** [comment]
Yes, I agree, this should work. Pull requests are always welcome!
